### PR TITLE
Bring Fedora image up to date.

### PR DESCRIPTION
**Pull request by jarrpa**
This is primarily for the purpose of uploading the image spec to the Fedora build system. Changes include:

* Update labels to meet Fedora requirements
* Dockerfile cleanup, remove outdated directives
* Add labels expected by Fedora build system
* Bring in brick multiplexing
* Fix bugs in existing scripts

Signed-off-by: Jose A. Rivera <jarrpa@redhat.com>

### CODE REVIEW

File: Fedora/gluster-setup.sh [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 ###
 # Description: Script to move the glusterfs initial setup to bind mounted directories of Atomic Host.
-# Copyright (c) 2016 Red Hat, Inc. <http://www.redhat.com>
+# Copyright (c) 2017 Red Hat, Inc. <http://www.redhat.com>
```
Comment (humblec) [inline Fedora/gluster-setup.sh]: This should be "2016" , why the change ?

File: Fedora/Dockerfile [hunk 2]
```diff
@@ -1,53 +1,62 @@
...
+      io.k8s.description="GlusterFS is a distributed and scalable network filesystem. Using common, off-the-shelf hardware you can create large, distributed storage solutions for media streaming, data analysis, and other data- and bandwidth-intensive tasks." \
+      io.openshift.tags="gluster,glusterfs,gluster-fedora" \
+      com.redhat.component="$NAME" \
+      maintainer="Jose A. Rivera <jarrpa@redhat.com>, Humble Chirammal <hchiramm@redhat.com>"
+
+VOLUME [ "/sys/fs/cgroup/" ]
+
+COPY gluster-setup.sh gluster-brickmultiplex.service gluster-brickmultiplex.sh gluster-setup.service /
+
+RUN dnf --nogpgcheck -y update && \
```
Comment (humblec) [inline Fedora/Dockerfile]: Lets install fedora gpg key and remove `--nogpgcheck`

File: Fedora/Dockerfile [hunk 3]
```diff
@@ -1,53 +1,62 @@
...
+    sed -i "s/LANG/\#LANG/g" /etc/locale.conf && \
+    dnf -y install --nogpgcheck systemd-udev ntp glusterfs-server dbus-python nfs-utils attr iputils iproute glusterfs-geo-replication openssh-server openssh-clients cronie tar rsync sos sudo xfsprogs && \
+    dnf clean all && \
+    sed -i '/Port 22/c\Port 2222' /etc/ssh/sshd_config && \
+    sed -i 's/Requires\=rpcbind\.service//g' /usr/lib/systemd/system/glusterd.service && \
+    sed -i 's/rpcbind\.service/gluster-setup\.service/g' /usr/lib/systemd/system/glusterd.service && \
+    sed -i.save -e "s#udev_sync = 1#udev_sync = 0#" -e "s#udev_rules = 1#udev_rules = 0#" -e "s#use_lvmetad = 1#use_lvmetad = 0#" /etc/lvm/lvm.conf && \
+    mkdir -p /etc/glusterfs_bkp /var/lib/glusterd_bkp /var/log/glusterfs_bkp && \
```
Comment (humblec) [inline Fedora/Dockerfile]: Better introduce comments on these files, so that its easy to know what we are doing on each layer.

**Comment by jarrpa** [comment]
@humblec Comments addressed! New commit pushed. :)

**Comment by humblec** [comment]
@jarrpa Thanks. Merging it
